### PR TITLE
Cleanup security procedure context

### PR DESCRIPTION
**Pull request by craigtaverner**
Refactoring of community security procedure:
- Port changePassword() to external API
- Remove AccessMode from NonTransactionalDbmsOperations (use only AuthSubject)
- Procedure exceptions with status should maintain status codes

changelog: Improve procedure context - provide only AuthSubject, use only exceptions with status codes

### CODE REVIEW

File: community/kernel/src/main/java/org/neo4j/kernel/impl/factory/ProceduresProvider.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.kernel.impl.factory;
```
Comment (fickludd) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/factory/ProceduresProvider.java:20]: It seems a little bit awkward to have this interface under `org.neo4j.kernel.impl.factory`. As alternatives, what about `org.neo4j.kernel.impl.proc` or `org.neo4j.kernel.api.proc` ?

File: community/kernel/src/test/java/org/neo4j/kernel/impl/api/integrationtest/BuiltinProceduresIT.java [hunk 2]
```diff
@@ -191,39 +190,18 @@ public void failWhenCallingListProceduresInDbmsMode() throws Throwable
...
-                .procedureCallDbms( procedureName( "dbms", "changePassword" ), inputArray, authSubject );
-
-        // Then
-        verify( authSubject ).setPassword( (String) inputArray[0] );
-        assertThat( asList( stream ), emptyIterable() );
-    }
-
-    @Test
-    public void shouldFailWhenChangePasswordWithStaticAccessModeInDbmsMode() throws Throwable
+    public void failWhenCallingNonExistingProcedures() throws Throwable
```
Comment (petraselmer) [inline community/kernel/src/test/java/org/neo4j/kernel/impl/api/integrationtest/BuiltinProceduresIT.java:193]: `failWhenCallingNonExistingProcedures` -> `failWhenCallingNonExistentProcedures`

File: community/security/src/test/java/org/neo4j/server/security/auth/AuthProceduresTest.java [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+import static junit.framework.TestCase.fail;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.emptyIterable;
+import static org.hamcrest.core.IsEqual.equalTo;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.neo4j.helpers.collection.Iterators.asList;
+import static org.neo4j.kernel.api.proc.ProcedureSignature.procedureName;
+
+public class AuthProceduresTest
```
Comment (fickludd) [inline community/security/src/test/java/org/neo4j/server/security/auth/AuthProceduresTest.java:43]: This should be UserProceduresTest?

File: community/kernel/src/main/java/org/neo4j/kernel/impl/factory/ProceduresProvider.java (hunk 1 above)
Comment (craigtaverner) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/factory/ProceduresProvider.java:20]: We thought about that too, but this interface is part of the factory mechanism for injection procedures, and so this still felt like the most natural place for the interface.

File: community/kernel/src/test/java/org/neo4j/kernel/impl/api/integrationtest/BuiltinProceduresIT.java (hunk 2 above)
Comment (craigtaverner) [inline community/kernel/src/test/java/org/neo4j/kernel/impl/api/integrationtest/BuiltinProceduresIT.java:193]: We're fixing this as part of a later PR.

File: community/security/src/test/java/org/neo4j/server/security/auth/AuthProceduresTest.java (hunk 3 above)
Comment (craigtaverner) [inline community/security/src/test/java/org/neo4j/server/security/auth/AuthProceduresTest.java:43]: We're fixing this as part of a later PR. But doing the opposite, moving UserProcedures to AuthProcedures for higher consistency.

**Pull request merged**